File: mobile/lib/features/neighborhoods/data/models/neighborhood_score.dart
/// The 11 scoreable factors, exactly matching `NeighborhoodScore::FACTORS`
/// server-side, with the labels the directory/profile pages show.
const kNeighborhoodScoreFactors = {
  'transport_access': 'Transport access',
  'schools': 'Schools',
  'hospitals': 'Hospitals',
  'markets': 'Markets',
  'internet_availability': 'Internet availability',
  'road_quality': 'Road quality',
  'noise': 'Noise',
  'safety': 'Safety',
  'flood_risk': 'Flood risk',
  'rental_demand': 'Rental demand',
  'development_activity': 'Development activity',
};

class NeighborhoodScoreFactor {
  NeighborhoodScoreFactor({required this.key, required this.score, this.notes});

  factory NeighborhoodScoreFactor.fromJson(Map<String, dynamic> json) {
    return NeighborhoodScoreFactor(
      key: json['key'] as String,
      score: json['score'] as int,
      notes: json['notes'] as String?,
    );
  }

  final String key;
  final int score; // 0-10
  final String? notes;

  String get label => kNeighborhoodScoreFactors[key] ?? key;
}

/// Mirrors the `score` object on `NeighborhoodSummaryResource`/
/// `NeighborhoodProfileResource`. `factors` is only ever populated on the
/// detail endpoint — the directory list always returns `[]` here since it
/// doesn't eager-load that nested relation.
class NeighborhoodScore {
  NeighborhoodScore({
    required this.overallScore,
    required this.source,
    this.computedAt,
    this.factors = const [],
  });

  factory NeighborhoodScore.fromJson(Map<String, dynamic> json) {
    return NeighborhoodScore(
      overallScore: json['overall_score'] as int,
      source: json['source'] as String,
      computedAt: json['computed_at'] as String?,
      factors: (json['factors'] as List<dynamic>? ?? const [])
          .map((f) => NeighborhoodScoreFactor.fromJson(f as Map<String, dynamic>))
          .toList(growable: false),
    );
  }

  final int overallScore; // 0-10
  final String source; // e.g. admin_curated
  final String? computedAt;
  final List<NeighborhoodScoreFactor> factors;
}

File: backend/app/Models/NeighborhoodScore.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class NeighborhoodScore extends Model
{
    // The canonical list of factors admins can rate — kept here so seeders,
    // validation, and the frontend all agree on the same set.
    public const FACTORS = [
        'transport_access', 'schools', 'hospitals', 'markets', 'internet_availability',
        'road_quality', 'noise', 'safety', 'flood_risk', 'rental_demand', 'development_activity',
    ];

    protected $fillable = ['neighborhood_id', 'overall_score', 'source', 'computed_at'];

    protected function casts(): array
    {
        return [
            'neighborhood_id' => 'integer',
            'overall_score' => 'integer',
            'computed_at' => 'datetime',
        ];
    }

    public function neighborhood(): BelongsTo
    {
        return $this->belongsTo(Neighborhood::class);
    }

    public function factors(): HasMany
    {
        return $this->hasMany(NeighborhoodScoreFactor::class);
    }
}
